### PR TITLE
GLTF parsing

### PR DESCRIPTION
**Pull request by danwillm**
Allows for the ability to parse GLTF animation files for hand tracking.

### CODE REVIEW

File: include/DeviceProvider.h [hunk 1]
```diff
@@ -3,9 +3,10 @@
...
 #define _WINSOCKAPI_
 
-#include <openvr_driver.h>
+#include "openvr_driver.h"
 
 #include <memory>
+#include "Bones.h"
 
 #include "Communication/CommunicationManager.h"
 #include "DeviceConfiguration.h"
```
Comment (noob247365) [inline include/DeviceProvider.h:12]:
```suggestion
#include <memory>

#include "Bones.h"
#include "Communication/CommunicationManager.h"
#include "DeviceConfiguration.h"
```

File: include/DriverLog.h [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #pragma once
 
 #include <string>
-#include <openvr_driver.h>
+#include "openvr_driver.h"
```
Comment (noob247365) [inline include/DriverLog.h:9]:
```suggestion

#include "openvr_driver.h"
```

File: include/Util/Windows.h [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+#include <string>
```
Comment (noob247365) [inline include/Util/Windows.h:2]:
```suggestion
#pragma once

#include <string>
```

File: src/Calibration.cpp [hunk 4]
```diff
@@ -6,6 +6,8 @@
 #include "DriverLog.h"
 #include "Quaternion.h"
 
+static const char* c_poseSettingsSection = "pose_settings";
```
Comment (noob247365) [inline src/Calibration.cpp:9]: Ideally we'd only define these in one location, now it's in both the `DeviceProvider.cpp` file and here

File: src/Communication/BTSerialCommunicationManager.cpp [hunk 5]
```diff
@@ -1,27 +1,9 @@
 #include <Communication/BTSerialCommunicationManager.h>
 #include <string.h>
+#include "Util/Windows.h"
```
Comment (noob247365) [inline src/Communication/BTSerialCommunicationManager.cpp:3]:
```suggestion
#include <Communication/BTSerialCommunicationManager.h>

#include <string.h>

#include "Util/Windows.h"
```

File: src/Communication/SerialCommunicationManager.cpp [hunk 6]
```diff
@@ -1,29 +1,11 @@
-#include <Communication/SerialCommunicationManager.h>
+#include "Communication/SerialCommunicationManager.h"
+#include "Util/Windows.h"
+#include "DriverLog.h"
```
Comment (noob247365) [inline src/Communication/SerialCommunicationManager.cpp:3]:
```suggestion
#include "Communication/SerialCommunicationManager.h"

#include "DriverLog.h"
#include "Util/Windows.h"
```

File: src/DeviceProvider.cpp [hunk 7]
```diff
@@ -68,29 +56,31 @@ vr::EVRInitError DeviceProvider::Init(vr::IVRDriverContext* pDriverContext) {
...
+  std::string driverPath = GetDriverPath();
+
+  const std::string unwanted = "\\bin\\win64";
+  driverPath.erase(driverPath.find(unwanted), unwanted.length());
+
+
+  VRDeviceConfiguration_t leftConfiguration = GetDeviceConfiguration(vr::TrackedControllerRole_LeftHand);
+  VRDeviceConfiguration_t rightConfiguration = GetDeviceConfiguration(vr::TrackedControllerRole_RightHand);
+
+    std::shared_ptr<BoneAnimator> boneAnimator = std::make_shared<BoneAnimator>(driverPath + "\\resources\\anims\\glove_anim.glb");
```
Comment (noob247365) [inline src/DeviceProvider.cpp:69]:
```suggestion
  VRDeviceConfiguration_t leftConfiguration = GetDeviceConfiguration(vr::TrackedControllerRole_LeftHand);
  VRDeviceConfiguration_t rightConfiguration = GetDeviceConfiguration(vr::TrackedControllerRole_RightHand);
  
  std::string driverPath = GetDriverPath();

  const std::string unwanted = "\\bin\\win64";
  driverPath.erase(driverPath.find(unwanted), unwanted.length());

    std::shared_ptr<BoneAnimator> boneAnimator = std::make_shared<BoneAnimator>(driverPath + "\\resources\\anims\\glove_anim.glb");
```

File: src/Util/Windows.cpp [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  return message;
+}
+
+std::string GetDriverPath() {
+  char path[MAX_PATH];
+  HMODULE hm = NULL;
+
+  if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, (LPCSTR)&__ImageBase, &hm) == 0) {
+    DriverLog("GetModuleHandle failed, error: %c", GetLastErrorAsString().c_str());
+    return "";
```
Comment (noob247365) [inline src/Util/Windows.cpp:31]:
```suggestion
    return std::string();
```

File: src/Util/Windows.cpp [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  if (GetModuleFileName(hm, path, sizeof(path)) == 0) {
+    DriverLog("GetModuleFileName failed, error: %c", GetLastErrorAsString().c_str());
+    return "";
```
Comment (noob247365) [inline src/Util/Windows.cpp:36]:
```suggestion
    return std::string();
```

File: src/Util/Windows.cpp [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  std::string::size_type pos = std::string(path).find_last_of("\\/");
+  return std::string(path).substr(0, pos);
```
Comment (noob247365) [inline src/Util/Windows.cpp:40]:
```suggestion
  std::string pathString = std::string(path);
  return pathString.substr(0, pathString.find_last_of("\\/"));
```

**Review by noob247365 — CHANGES_REQUESTED** [review]

**Review by noob247365 — APPROVED** [review]

File: src/Calibration.cpp (hunk 4 above)
Comment (danwillm) [reply src/Calibration.cpp:9]: Resolved this in a later commit

**Review by noob247365 — APPROVED** [review]

**Review by lucas-vrtech — APPROVED** [review]

**Pull request merged**